docs/best_practices.md

```diff
@@ -0,0 +1,153 @@
+
+```
+buckets()
+  |> filter(fn: (r) => r.name !~ /^_/)
+  |> rename(columns: {name: "_value"})
+  |> keep(columns: ["_value"])
+```
+> **Note:** the `filter()` call removes the system Buckets `_monitoring` and `_tasks` from the list
+
+Then, in your Dashboard you can read from the user's chosen Bucket in a cell query like this:
+
+```
+from(bucket: v.bucket) 
+```
+
+Once a Variable is used in a Dashboard, it will be shown to the user at the top of the page as a drop-down menu where they can change which value to use for that Variable.
+
+![Bucket Variable](img/bucket_variable.png)
+
+> **Tip:** If you are including a Telegraf configuration in your Template, be sure to use an environment variable such as `$INFLUX_BUCKET` so the user can define which Bucket they want it to send data to.
+
+### Using Variables
+
+[Variables](https://v2.docs.influxdata.com/v2.0/visualize-data/variables/) are a great way to give users flexibility over your Template without them having to make any changes to it. In the previous section you saw how a Variable can be used to display a list of Buckets to use, but they can also provide static or dynamic options anywhere you use Flux. You can create a Variable fromthe `Settings` -> `Variables` page of the InfluxDB UI.
+
+![Create a Variable](img/create_variable.png)
+
+#### Static Variables
+
+There are two ways to create a static Variable. The first is to provide a simple list of options as comma-separated values (CSV):
+
+```
+value1,value2,value3
+```
+
+The other is to provide a map of values and display names:
+
+```
+display1:value1
+display2:value2
+display3:value3
+```
+
+#### Variables from tags
+
+A common use case is in InfluxDB is to filter data by one of the `tags` in the measurement. You can make this configurable in your Dashboard by making a Variable out of them. Create a Query Variable with a Flux query like this one:
+
+```
+import "influxdata/influxdb/v1"
+v1.measurementTagValues(bucket: "your_bucket_name", measurement: "your_measurement_name", tag: "your_tag_name")
+```
+
+This will give the user a drop-down menu with all of the unique values of the tag `your_tag_name` from the specified measurements.
+
+### Using Labels
+
+[Labels](https://v2.docs.influxdata.com/v2.0/visualize-data/labels/) let you tag your resources for easier identification in the UI, and they will also make it easier to export them into a Template. If you have a mix of resources in your InfluxDB instance you will have to individually specify which ones to export so that you don't get them all.
+
+However, if you create a unique Label to identify those resources, you can use that when exporting so that you don't have to know the IDs of each individual resource you want to include, like this:
```

Comment:
- This should be part of the paragraph above.
- However, if you ~create a unique Label~ **use unique labels** to identify ~those~ resources, you can ~use that~  **filter resources by label** when exporting ~so that you don't have to know the IDs of each individual resource you want to include, like this~ your template: